### PR TITLE
Implementation of Product Management Interface and Service

### PR DESCRIPTION
**Pull request by ei18**
Implementation of the interface and service necessary for product management, fulfilling the assigned user story. 

Creation of the IProductService interface extending the following generic interfaces. This interface defines the necessary methods for creating, reading, updating, and deleting products.

**ProductService Service Class:**
Implementation of the ProductService class that implements IProductService.
**Implemented methods:**

- create(ProductRequest request): Converts the ProductRequest to Product and saves the product in the repository.

- getById(String id): Retrieves a product by its ID and converts it to ProductResponse.

- update(String id, ProductRequest request): Updates an existing product based on the provided ID and ProductRequest.

- delete(String id): Deletes a product based on the ID.

- Auxiliary method find(String id): Finds a product by its ID and throws an exception if not found.

### CODE REVIEW

File: src/main/java/com/riwi/localstorage/riwi_local_storage/api/controllers/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import lombok.AllArgsConstructor;
+
+@RestController
+@RequestMapping(path = "/products")
+@AllArgsConstructor
+public class ProductController {
+    
+    private final IProductService productService;
+
+    @GetMapping(path = "/{id}")
```
Comment (cristianbyte) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/api/controllers/ProductController.java]: path = "/admin/{id}"

File: src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/services/ProductService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Override
+    public Optional<ProductResponse> getById(String id) {
+        
+        return productRepository.findById(id)
+                .map(productMapper::productToProductResponse);
+    }
+
+    @Override
+    public Page<ProductResponse> getAll(Pageable pageable) {
```
Comment (cristianbyte) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/services/ProductService.java:44]: still missing

**Review by cristianbyte — CHANGES_REQUESTED** [review]

**Comment by ei18** [comment]
**Changes Implemented:**
The following methods were implemented in the service and repository:

**In ProductRepository:**

Page<Product> findAllByIsEnableTrue(Pageable pageable);
Optional<Product> findByIdAndIsEnableTrue(String id);
These methods allow fetching paginated enabled products and finding enabled products by ID, respectively.

**In ProductService:**

ProductResponse create(ProductRequest request): Creates a new enabled product.
Optional<ProductResponse> getById(String id): Retrieves a product by ID if it's enabled.
Page<ProductResponse> getAll(Pageable pageable): Retrieves all enabled products in a paginated manner.
ProductResponse update(String id, ProductRequest request): Updates an existing enabled product.
void delete(String id): Disables an existing product by changing its enablement state.
Additionally, a find(String id) method was added to retrieve a product by ID if enabled or throw an IdNotFoundException if not found.

**In ProductController:**

RESTful endpoints were implemented to handle basic CRUD operations (GET, POST, PUT, DELETE) for products.
Additional Details:
@Transactional was used to ensure transactional integrity for operations.
@Validated annotations were applied for input validation in operations receiving data (create and update).
DTO (Data Transfer Object) pattern was employed with ProductRequest and ProductResponse for data transfer between controller, service, and repository layers.
Mappers (ProductMapper) were integrated to convert between DTOs and domain entities.

This code has been tested and its functionality verified prior to submitting this pull request.

**Review by cristianbyte — COMMENTED** [review]
well done

**Pull request merged**